src/main/java/com/google/api/codegen/discovery/transformer/java/JavaSampleTypeNameConverter.java

```diff
@@ -67,84 +64,39 @@
-  /**
-   * Upper-case conversion of methodNameComponents from SampleConfig.
-   */
-  private final List<String> methodNameComponents;
-
-  public JavaSampleTypeNameConverter(SampleConfig sampleConfig, String methodName) {
-    // We store these values to facilitate import path generation.
+  public JavaSampleTypeNameConverter(String packagePrefix) {
     this.typeNameConverter = new JavaTypeTable();
-    String apiTypeNameOverride = sampleConfig.apiTypeNameOverride();
-    if (!Strings.isNullOrEmpty(apiTypeNameOverride)) {
-      this.apiTypeName = typeNameConverter.getTypeName(apiTypeNameOverride).getNickname();
-    } else {
-      this.apiTypeName = Name.lowerCamel(sampleConfig.apiName()).toUpperCamel();
-    }
-    this.apiNameVersion = sampleConfig.apiNameVersion();
-    List<String> copy = new ArrayList<>(sampleConfig.methods().get(methodName).nameComponents());
-    // Convert the method name components to upper camel case.
-    for (int i = 0; i < copy.size(); i++) {
-      copy.set(i, Name.lowerCamel(copy.get(i)).toUpperCamel());
-    }
-    this.methodNameComponents = copy;
+    this.packagePrefix = packagePrefix;
   }
 
   @Override
-  public TypeName getServiceTypeName(SampleConfig sampleConfig) {
-    return typeNameConverter.getTypeName(
-        String.join(".", "com.google.api.services", apiNameVersion, apiTypeName));
+  public TypeName getServiceTypeName(String apiTypeName) {
+    return typeNameConverter.getTypeName(Joiner.on('.').join(packagePrefix, apiTypeName));
   }
 
-  /**
-   * Returns a TypeName from a request message type.
-   *
-   * Since request imports are special in Java, in that they have a different
-   * prefix, the logic is separate from the standard flow.
-   */
   @Override
-  public TypeName getRequestTypeName(TypeInfo typeInfo) {
-    // By default, use the concatenation of the method name components. If the
-    // default type name is overridden, use that value instead.
-    String typeName = typeInfo.message().typeName();
-    if (typeName.equals(DiscoveryImporter.REQUEST_FIELD_NAME)) {
-      typeName = String.join(".", methodNameComponents);
-    }
-    return getTypeName(
-        typeInfo,
-        String.join(".", "com.google.api.services", apiNameVersion, apiTypeName, typeName));
+  public TypeName getRequestTypeName(String apiTypeName, String requestTypeName) {
+    return typeNameConverter.getTypeName(
+        Joiner.on('.').join(packagePrefix, apiTypeName, requestTypeName));
   }
 
-  /**
-   * Returns a TypeName from typeInfo.
-   */
   @Override
   public TypeName getTypeName(TypeInfo typeInfo) {
-    String typeName = "";
     if (typeInfo.isMessage()) {
-      typeName =
-          String.join(
-              ".",
-              "com.google.api.services",
-              apiNameVersion,
-              "model",
-              typeInfo.message().typeName());
+      return typeNameConverter.getTypeName(
+          Joiner.on('.').join(packagePrefix, "model", typeInfo.message().typeName()));
```

Comment:
Maybe make a subpackage attribute in TypeInfo, and set it to "model" earlier in the process, so that it doesn't have to be encoded here.